src/ol/interaction/ModifyPolygonBrush.js

```diff
@@ -0,0 +1,283 @@
+  /**
+   * @param {Options} options Options.
+   */
+  constructor(options) {
+
+    super(/** @type {import("./Pointer.js").Options} */ (options));
+
+    this.sketchFeature_ = null;
+
+    this.modifyFeature_ = null;
+
+    this.sketchPoint_ = null;
+
+    this.mode_ = options.mode ? options.mode : 'add';
+
+    /**
+     * Draw overlay where sketch features are drawn.
+     * @type {VectorLayer}
+     * @private
+     */
+    this.overlay_ = new VectorLayer({
+      source: new VectorSource({
+        useSpatialIndex: false,
+        wrapX: options.wrapX ? options.wrapX : false
+      }),
+      style: options.style ? options.style :
+        getDefaultStyleFunction(),
+      updateWhileInteracting: true
+    });
+
+    /**
+     * @type {VectorSource}
+     * @private
+     */
+    this.source_ = null;
+
+    /**
+     * @type {import("../MapBrowserPointerEvent.js").default}
+     * @private
+     */
+    this.lastPointerEvent_ = null;
+
+    this.circleRadius_ = 10000;  //TODO better value
+    this.drawmode_ = false;
+
+  }
+
+  /**
+   * Redraw the sketch features.
+   * @private
+   */
+  updateSketchFeatures_() {
+    const sketchFeatures = [];
+    if (this.sketchFeature_) {
+      sketchFeatures.push(this.sketchFeature_);
+    }
+    if (this.sketchLine_) {
+      sketchFeatures.push(this.sketchLine_);
+    }
+    if (this.sketchPoint_) {
+      sketchFeatures.push(this.sketchPoint_);
+    }
+    const overlaySource = /** @type {VectorSource} */ (this.overlay_.getSource());
+    overlaySource.clear(true);
+    overlaySource.addFeatures(sketchFeatures);
+  }
+
+  /**
+   * @param {import("../MapBrowserEvent.js").default} event Event.
+   * @private
+   */
+  createOrUpdateSketchPoint_(event) {
+    const coordinates = event.coordinate.slice();
+    if (!this.sketchPoint_) {
+      this.sketchPoint_ = new Feature(new Circle(coordinates,this.circleRadius_));
+      this.updateSketchFeatures_();
+    } else {
+      const sketchPointGeom = /** @type {Circle} */ (this.sketchPoint_.getGeometry());
+      sketchPointGeom.setCenter(coordinates);
+    }
+  }
+
+  updateSketchPointRadius_(event) {
+    if (this.sketchPoint_) {
+      const sketchPointGeom = /** @type {Circle} */ (this.sketchPoint_.getGeometry());
+      if (event.originalEvent.deltaY > 0) {
+        this.circleRadius_ = sketchPointGeom.getRadius() + 1000;  //TODO better value
+      }
+      if (event.originalEvent.deltaY < 0) {
+        this.circleRadius_ = sketchPointGeom.getRadius() - 1000;  //TODO better value
+      }
+      sketchPointGeom.setRadius(this.circleRadius_);
+    }
+  }
+
+  handleEvent(event) {
+    let pass = true;
+    const type = event.type;
+    const btn = event.originalEvent.button;
+    if (shiftKeyOnly(event) && (type === EventType.WHEEL || type === EventType.MOUSEWHEEL)) {
+      pass = false;
+      this.updateSketchPointRadius_(event);
+    }
+    if (btn == 0 && (type === MapBrowserEventType.POINTERDOWN)) {
```

Comment:
In den meisten Fällen ist es besser, wenn man `===` statt `==` verwendet. Das würde ich hier auch so halten.